### PR TITLE
feat(core): Apiclientfactory

### PR DESCRIPTION
**Pull request by filrak**
- [x] I Accept [Individual Contributor License Agreement](https://github.com/DivanteLtd/next/blob/master/CLA.md)

### CODE REVIEW

File: packages/core/factories/src/apiClientFactory.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export function apiClientFactory<API_CLIENT_METHODS, API_CLIENT_SETTINGS>(factoryParams: { defaultSettings; onSetup }) {
+  let settings = { ...factoryParams.defaultSettings };
```
Comment (andrzejewsky) [inline packages/core/factories/src/apiClientFactory.ts]:
```
let settings = { ...factoryParams.defaultSettings, overrides: {} };
```

**Review by andrzejewsky — CHANGES_REQUESTED** [review]
looks ok 👍 just one comment and look what's wrong with the build

File: packages/core/factories/src/apiClientFactory.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    },
+    setup (config: API_CLIENT_SETTINGS) {
+      settings = { ...factoryParams.defaultSettings, overrides: settings.overrides, ...config };
+      factoryParams.onSetup(settings);
+    },
+    update (config: API_CLIENT_SETTINGS) {
+      settings = { ...settings, ...config };
+      factoryParams.onSetup(settings);
+    },
+    getSettings: () => Object.freeze(settings)
```
Comment (patzick) [inline packages/core/factories/src/apiClientFactory.ts]: this should return type with `API_CLIENT_SETTINGS ` for IDE

**Review by patzick — APPROVED** [review]
I'd need to see the use case for this factory, but overall looks okay to me :)

**Review by andrzejewsky — APPROVED** [review]

**Pull request merged**